tests/BlockTypeRegistry/BlockStateTest.cpp

```diff
@@ -109,8 +109,66 @@ static void testReplacing()
 
 
 
+/** Tests the comparison operator. */
+static void testComparison()
+{
+	LOGD("Testing comparison of BlockStates...");
+
+	// Simple property value tests
+	TEST_FALSE((BlockState({{"a","a"}}) < BlockState({{"a","a"}})));  // equal
+	TEST_FALSE((BlockState({}) < BlockState({})));  // equal
+	TEST_TRUE((BlockState({}) < BlockState({{"foo","bar"}})));
+	TEST_FALSE((BlockState({{"foo","bar"}}) < BlockState({})));
+}
+
+
+
+
+
+/** Tests the comparison operator using crafted data to defeat the checksum. */
+static void testComparison2()
+{
+	/* The following test ensures that items inserted in different order result
+	in the same map. I.e. that the < operator is stable. */
+	std::vector<BlockState> v;
+	std::map<BlockState, bool> map1;
+	std::map<BlockState, bool> map2;
+
+	for (int i=0; i < 128; ++i)
```

Comment:
... I'm sorry about all the code style, yours almost directly contradicts my normal one, that's why I don't notice - also because the pre-commit hook doesn't catch them...

Reply:
Unfortunately it's rather difficult to catch these without any false positives, if you can improve the precommit hook (it's Lua, after all), that'd be great.

Reply:
Btw. I'm not sure now about unary operators - so `++i` or `++ i`? I found only `++i` in existing code though CONTRIBUTING says `every operator` without qualification.

> if you can improve the precommit hook

It's Lua, but it checks C++ syntax, Pretty sure to fully check your rules a tokenizer would be required, then it's possible to distinguish between unary and binary operators, comments or no comments, etc.

Reply:
Good point, I'll clarify that there. Unary operators without a space.

Yes, a tokenizer is needed for full support, but perhaps some more could still be caught by simple Lua regexes.

Reply:
(done)

> I'll clarify that there

Maybe check the other rules too, e.g. the method `loadFromString` I think would not be allowed (functions start with uppercase), classes start with `c`, members start with `m_` (underline!), etc.

Reply:
That's temporary, with the 1.13 changes we plan to change that rule, so we're already using new style for new code, but keeping old style in old code.
(4th paragraph in first post of https://forum.cuberite.org/thread-3215.html )